File: CourseWorkOnAndroidAndSpring-main/project/server/src/main/java/ru/ryabov/studentperformance/dto/grade/GradeDto.java
package ru.ryabov.studentperformance.dto.grade;

import java.math.BigDecimal;
import java.time.LocalDate;

public class GradeDto {

    private Long gradeId;
    private Long studentId;
    private String studentFullName;
    private Long subjectId;
    private String subjectName;
    private String gradeTypeName;
    private BigDecimal gradeValue;
    private LocalDate gradeDate;
    private Integer semester;
    private Integer academicYear;
    private String comment;
    private String workType;

    public GradeDto() {
    }

    public GradeDto(Long gradeId, Long studentId, String studentFullName, Long subjectId,
                    String subjectName, String gradeTypeName, BigDecimal gradeValue,
                    LocalDate gradeDate, Integer semester, Integer academicYear,
                    String comment, String workType) {
        this.gradeId = gradeId;
        this.studentId = studentId;
        this.studentFullName = studentFullName;
        this.subjectId = subjectId;
        this.subjectName = subjectName;
        this.gradeTypeName = gradeTypeName;
        this.gradeValue = gradeValue;
        this.gradeDate = gradeDate;
        this.semester = semester;
        this.academicYear = academicYear;
        this.comment = comment;
        this.workType = workType;
    }

    // Getters and Setters
    public Long getGradeId() { return gradeId; }
    public void setGradeId(Long gradeId) { this.gradeId = gradeId; }

    public Long getStudentId() { return studentId; }
    public void setStudentId(Long studentId) { this.studentId = studentId; }

    public String getStudentFullName() { return studentFullName; }
    public void setStudentFullName(String studentFullName) { this.studentFullName = studentFullName; }

    public Long getSubjectId() { return subjectId; }
    public void setSubjectId(Long subjectId) { this.subjectId = subjectId; }

    public String getSubjectName() { return subjectName; }
    public void setSubjectName(String subjectName) { this.subjectName = subjectName; }

    public String getGradeTypeName() { return gradeTypeName; }
    public void setGradeTypeName(String gradeTypeName) { this.gradeTypeName = gradeTypeName; }

    public BigDecimal getGradeValue() { return gradeValue; }
    public void setGradeValue(BigDecimal gradeValue) { this.gradeValue = gradeValue; }

    public LocalDate getGradeDate() { return gradeDate; }
    public void setGradeDate(LocalDate gradeDate) { this.gradeDate = gradeDate; }

    public Integer getSemester() { return semester; }
    public void setSemester(Integer semester) { this.semester = semester; }

    public Integer getAcademicYear() { return academicYear; }
    public void setAcademicYear(Integer academicYear) { this.academicYear = academicYear; }

    public String getComment() { return comment; }
    public void setComment(String comment) { this.comment = comment; }

    public String getWorkType() { return workType; }
    public void setWorkType(String workType) { this.workType = workType; }
}

File: CourseWorkOnAndroidAndSpring-main/project/mobile/app/src/main/java/ru/zagrebin/studentperformance/data/remote/dto/GradeDto.kt
package ru.ryabov.studentperformance.data.remote.dto

import com.google.gson.annotations.SerializedName
import java.math.BigDecimal

data class GradeDto(
    @SerializedName("gradeId") val gradeId: Long,
    @SerializedName("studentId") val studentId: Long,
    @SerializedName("studentFullName") val studentFullName: String? = null,
    @SerializedName("subjectId") val subjectId: Long,
    @SerializedName("subjectName") val subjectName: String? = null,
    @SerializedName("gradeTypeName") val gradeTypeName: String? = null,
    @SerializedName("gradeValue") val gradeValue: BigDecimal? = null,
    @SerializedName("gradeDate") val gradeDate: String? = null,
    @SerializedName("semester") val semester: Int? = null,
    @SerializedName("academicYear") val academicYear: Int? = null,
    @SerializedName("comment") val comment: String? = null,
    @SerializedName("workType") val workType: String? = null
)

data class CreateGradeRequest(
    @SerializedName("studentId") val studentId: Long,
    @SerializedName("subjectId") val subjectId: Long,
    @SerializedName("assignmentId") val assignmentId: Long? = null,
    @SerializedName("gradeTypeId") val gradeTypeId: Long? = null,
    @SerializedName("gradeValue") val gradeValue: BigDecimal? = null,
    @SerializedName("gradeDate") val gradeDate: String? = null,
    @SerializedName("semester") val semester: Int? = null,
    @SerializedName("academicYear") val academicYear: Int? = null,
    @SerializedName("comment") val comment: String? = null,
    @SerializedName("workType") val workType: String? = null
)

data class GradeSummaryDto(
    @SerializedName("averageGrade") val averageGrade: java.math.BigDecimal? = null,
    @SerializedName("totalGrades") val totalGrades: Int = 0,
    @SerializedName("excellentCount") val excellentCount: Int = 0,
    @SerializedName("goodCount") val goodCount: Int = 0,
    @SerializedName("satisfactoryCount") val satisfactoryCount: Int = 0,
    @SerializedName("failCount") val failCount: Int = 0
)
